### ISSUE TITLE
process_type 'update' causes kernel crash for classification problem

### ISSUE THREAD

**Issue by charles882** [dmlc/xgboost#2315]
## Environment info
Operating System: Windows 10

Compiler: build with updater_gpu plugin, using visual studio 2015, no error

Package used (python/R/jvm/C++): Python 

`xgboost` version used: 0.6

If installing from source, please provide 

1. The commit hash (`git rev-parse HEAD`)  a607f697e37035a16a9d42496ec2b5f2e1affe41

If you are using python package, please provide

1. The python version and distribution: python 2.7.12 64 bit, anaconda for windows 

## Steps to reproduce
1. I tried to do incremental learning with classification problem but kernel crashed.
setting updater to 'refresh' and process_type 'update' causes kernel crash. This code reproduce the problem:

import xgboost as xgb
import numpy as np
import random
random.seed(1)
from sklearn.metrics import mean_squared_error as mse

x = np.reshape(np.random.randint(1,100,size = 600),(60,-1))
y = np.reshape(np.random.randint(0,3,size = 60),(60,-1))

xg_train_0 = xgb.DMatrix(x[:30], label=y[:30])
xg_train_1 = xgb.DMatrix(x[30:], label=y[30:])

params = {'objective':'multi:softprob', 'verbose': False, 'num_class': 3} #classification
#params = {'objective': 'reg:linear', 'verbose': False}
model_0 = xgb.train(params, xg_train_0, 10)

params.update({'process_type': 'update',
               'updater'     : 'refresh',
               'refresh_leaf': True})
model_1 = xgb.train(params, xg_train_1, 10, xgb_model=model_0)

2.
3.

## What have you tried?

1. using parameter {'process_type': 'update', 'updater': 'refresh','refresh_leaf': True} for classification causes crash
2. using parameter {'updater': 'refresh','refresh_leaf': True} works but my understanding is that this runs the default boosting process instead of update process. From the documentation:
https://github.com/dmlc/xgboost/blob/master/doc/parameter.md

process_type, [default='default']
A type of boosting process to run.
Choices: {'default', 'update'}
'default': the normal boosting process which creates new trees.
'update': starts from an existing model and only updates its trees. In each boosting iteration, a tree from the initial model is taken, a specified sequence of updater plugins is run for that tree, and a modified tree is added to the new model. The new model would have either the same or smaller number of trees, depending on the number of boosting iteratons performed. Currently, the following built-in updater plugins could be meaningfully used with this process type: 'refresh', 'prune'. With 'update', one cannot use updater plugins that create new nrees.

3. using parameter {'process_type': 'update', 'updater': 'refresh','refresh_leaf': True} on regression problem works.

Should the update process works for classification problem as well? I wanted to have faster learning to deal with new data without having to re-learn whole data from scratch.

Any help is appreciated. Thanks in advance

**Comment by khotilov** [comment]
Thanks for the report. Please check the fix in #2332.

**Comment by charles882** [comment]
The fix works, I am using it now. It does not crash anymore when using process_type 'update' for classification problem. Thank you!
